plugins/database/mysql/mysql.go

```diff
@@ -316,6 +316,89 @@ func (m *MySQL) RotateRootCredentials(ctx context.Context, statements []string)
 	return m.RawConfig, nil
 }
 
+// SetCredentials uses provided information to set the password a user in the
+// database. Unlike CreateUser, this method requires a username be provided and
+// uses the name given, instead of generating a name. This is used for setting
+// the password of static accounts, as well as rolling back passwords in the
+// database in the event an updated database fails to save in Vault's storage.
+func (m *MySQL) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	if len(statements.Rotation) == 0 {
+		return "", "", errors.New("empty rotation statements")
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	// Grab the lock
+	m.Lock()
+	defer m.Unlock()
+
+	// Get the connection
+	db, err := m.getConnection(ctx)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Start a transaction
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return "", "", err
+	}
+	defer func() {
+		_ = tx.Rollback()
+	}()
+	// Return the secret
+
+	stmts := statements.Rotation
+	// Execute each query
+	for _, stmt := range stmts {
+		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+			query = strings.TrimSpace(query)
+			if len(query) == 0 {
+				continue
+			}
+			query = dbutil.QueryHelper(query, map[string]string{
```

Comment:
For my own edification, why are we doing string substitution on the template, rather than having everything be SQL parameters?

I mean, I don't expect there's a way to do SQL injection but I would be concerned about whether the password is escaped correctly if it contains odd characters.

Reply:
Unfortunately I don't know the answer to this, nor do I know `database/sql` well enough to comment. I believe "all" of the generated passwords currently come from a random alphanumeric funciton and don't contain special characters